flink-cdc-common/src/main/java/org/apache/flink/cdc/common/schema/Selectors.java

```diff
@@ -29,8 +33,16 @@
 /** Selectors for filtering tables. */
 public class Selectors {
 
+    private static final Duration CACHE_EXPIRE_DURATION = Duration.ofHours(1);
+
     private List<Selector> selectors;
 
+    private final Cache<TableId, Boolean> cache =
+            CacheBuilder.newBuilder()
+                    .expireAfterAccess(CACHE_EXPIRE_DURATION)
+                    .maximumSize(1024)
+                    .build();
```

Comment:
Is it correct to share this cache among all `Selector`s? IIUC it depends on the effective `Selector` to determine if a `TableId` should be matched.